waddrmgr/manager_test.go

```diff
@@ -121,6 +138,12 @@ func testManagedPubKeyAddress(tc *testContext, prefix string, gotAddr waddrmgr.M
 	// Since this is only available when the manager is unlocked, also check
 	// for the expected error when the manager is locked.
 	gotPrivKey, err := gotAddr.PrivKey()
+	if wantAddr.watchingOnly {
```

Comment:
I'd combine this in the switch case below like so:

```Go
case tc.watchingOnly, wantAddr.watchingOnly:
``` 
Can avoid the extra check and get the better error prefix on a failure.